akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala

```diff
@@ -159,15 +159,13 @@ private[persistence] trait Eventsourced extends Snapshotter with Stash with Stas
       toSequenceNr, persistenceId, cause.getMessage)
   }
 
-  /**
-   * User-overridable callback. Called when a persistent actor is started or restarted.
-   * Default implementation sends a `Recover()` to `self`. Note that if you override
-   * `preStart` (or `preRestart`) and not call `super.preStart` you must send
-   * a `Recover()` message to `self` to activate the persistent actor.
-   */
-  @throws(classOf[Exception])
-  override def preStart(): Unit =
-    self ! Recover()
+  override def preStart() =
+    startRecovery(recovery) // TODO try moving to aroundPreStart
+
+  private def startRecovery(recovery: Recover): Unit = {
+    changeState(recoveryStarted(recovery.replayMax))
+    loadSnapshot(snapshotterId, recovery.fromSnapshot, recovery.toSequenceNr)
+  }
```

Comment:
can't we remove the `recoveryPending` state, and start with `currentState = recoveryStarted`?

and, yes, should be done in `aroundPreStart`. what is the problem?